tests/unit/test_flight.py

```diff
@@ -957,3 +957,72 @@ def test_resample_and_fill_two_waypoints() -> None:
     # variation in groundspeed than geodesic interpolation.
     gs = fl.segment_groundspeed()[:-1]
     np.testing.assert_allclose(gs, 210.0, atol=1.0)
+
+
+def test_flight_to_dict(flight_fake: Flight) -> None:
+    """Test the Flight.to_dict method."""
+    fl = flight_fake.copy()
+
+    # modify lon to be more precise
+    fl.update({"longitude": 0.999 * fl["longitude"]})
+
+    # Add some additional attributes
+    fl.attrs["aircraft_type"] = "A320"
+    fl.attrs["flight_number"] = "BA123"
+    fl.attrs["some_time"] = pd.Timestamp("2021-01-01 00:00:00")
+    fl.attrs["some_duration"] = pd.Timedelta("1H")
+
+    flight_dict = fl.to_dict()
+    assert isinstance(flight_dict, dict)
+    for k in flight_dict:
+        assert isinstance(k, str)
+
+    assert flight_dict["aircraft_type"] == "A320"
+    assert flight_dict["flight_number"] == "BA123"
+
+    # Only the altitude_ft column is included
+    assert "altitude_ft" in flight_dict
+    assert "altitude" not in flight_dict
+    assert "level" not in flight_dict
+
+    # Ensure serializable
+    assert json.dumps(flight_dict)
+
+    # Ensure time is in unix seconds
+    assert flight_dict["time"][0] < 1e10
+
+    # Ensure that lat/lon have 3 decimals
+    assert len(str(flight_dict["latitude"][0]).split(".")[1]) <= 3
+    assert len(str(flight_dict["longitude"][0]).split(".")[1]) <= 3
+
+    # Ensure alt has 1 decimal
+    assert len(str(flight_dict["altitude_ft"][0]).split(".")[1]) == 1
+
+
+def test_flight_to_dict_conflicts(flight_fake: Flight) -> None:
+    """Test the Flight.to_dict method with conflicting keys."""
+
+    flight_fake["engine_efficiency"] = np.linspace(0.2, 0.4, len(flight_fake))
+    flight_fake.attrs["engine_efficiency"] = 0.33
+
+    match = "Found duplicate keys in data and attrs: {'engine_efficiency'}"
+    with pytest.warns(UserWarning, match=match):
+        flight_dict = flight_fake.to_dict()
+
+    ee = flight_dict["engine_efficiency"]
+    assert isinstance(ee, list)
+    np.testing.assert_array_equal(ee, flight_fake["engine_efficiency"])
+
+    # Ensure serializable
+    assert json.dumps(flight_dict)
+
+
+def test_flight_from_dict(flight_fake: Flight) -> None:
```

Comment:
One more test you could add would check what happens when you compose `to_dict` with `from_dict`. Ideally we'd want

```python
assert fl == Flight.from_dict(flight.to_dict())
```

to hold. Due to the rounding we employ and not converting attrs back to numpy in the `from_dict` method, this will probably be too strict ... but it should hold in some examples.

Reply:
Good idea, the test is a little contrived, but its a helpful exercise 4fb6eb497c79c95b98d8ab3cc1601590fdf1b190